std/parallelism.d

```diff
@@ -2864,10 +2864,10 @@ public:
             }
         }
 
-        ref T opIndex(size_t index)
+        ref inout(T) opIndex(size_t index) inout
```

Comment:
`inout` and `this Q` are overlapping but not identical features with overlapping but not identical pros and cons. Of these, on the whole `this Q` is the true enabler whereas `inout` is a complicated mechanism of limited power. I wouldn't disagree that `inout` has advantages in certain situations, but on the whole we'd do good to stick with fewer overlapping ways of doing things.